test/plots/rectanglePlotTests.ts

```diff
@@ -2,444 +2,418 @@
-        .x((d) => d.x, xScale).x2((d) => d.x2)
-        .y((d) => d.y, yScale).y2((d) => d.y2);
-      plot.addDataset(dataset).renderTo(svg);
-
-      let entities = plot.entitiesIn(
-        {min: xScale.scale(1.5), max: xScale.scale(2.5)},
-        {min: yScale.scale(2.5), max: yScale.scale(1.5)});
-      assert.lengthOf(entities, 2, "retrieved 2 entities intersect with the box");
-      assert.strictEqual(entities[0].index, 1, "the entity of index 1 is retrieved");
-      assert.strictEqual(entities[1].index, 2, "the entity of index 2 is retrieved");
-      svg.remove();
-    });
+        VERIFY_CELLS((<any> plot)._renderArea.selectAll("rect"));
+      });
 
-    it("autorangeMode(\"x\")", () => {
-      let staggeredData = [
-        { y: "A", x: 0, x2: 1 },
-        { y: "B", x: 1, x2: 2 }
-      ];
+      it("retrieves the correct entity under a point", () => {
+        plot.addDataset(new Plottable.Dataset(DATA));
+        plot.renderTo(svg);
 
-      let xScale = new Plottable.Scales.Linear();
-      let yScale = new Plottable.Scales.Category();
-      xScale.padProportion(0);
+        let entities = plot.entitiesAt({ x: xScale.scale(2.5), y: yScale.scale(2.5) });
+        assert.lengthOf(entities, 1, "found only one entity when querying a point inside the third rectangle");
+        assert.strictEqual(entities[0].index, 2, "entity retrieved is at index 2");
+      });
 
-      let plot = new Plottable.Plots.Rectangle();
-      plot.x(function(d) { return d.x; }, xScale);
-      plot.x2(function(d) { return d.x2; });
-      plot.y(function(d) { return d.y; }, yScale);
-      plot.addDataset(new Plottable.Dataset(staggeredData));
-      plot.autorangeMode("x");
-      let svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
-      plot.renderTo(svg);
+      it("retrieves correct entities under a point", () => {
+        let dataset = new Plottable.Dataset([
+          { x: 1, y: 1, x2: 3, y2: 3 },
+          { x: 4, y: 2, x2: 2, y2: 4 }
+        ]);
+        plot.addDataset(dataset);
+        plot.renderTo(svg);
+
+        let entities = plot.entitiesAt({ x: xScale.scale(2), y: xScale.scale(2) });
+        assert.lengthOf(entities, 2, "two entities when querying a point in both");
+        assert.strictEqual(entities[0].index, 0, "entity retrieved is at index 0");
+        assert.strictEqual(entities[1].index, 1, "entity retrieved is at index 1");
+        entities = plot.entitiesAt({ x: xScale.scale(4), y: yScale.scale(4) });
+        assert.lengthOf(entities, 1, "found only one entity when querying a point inside the second rectangle");
+        assert.strictEqual(entities[0].index, 1, "entity retrieved is at index 1");
+        entities = plot.entitiesAt({ x: xScale.scale(1), y: yScale.scale(1) });
+        assert.lengthOf(entities, 1, "found only one entity when querying a point inside the first rectangle");
+        assert.strictEqual(entities[0].index, 0, "entity retrieved is at index 0");
+      });
```

Comment:
Not really. For the above test, the rectangles did not overlap (which is the common case for a rectangle plot). For this one they do. Since this is what the test is testing, I amended the title of the test only and left both tests in